src/utils.cpp

```diff
@@ -514,6 +514,18 @@ int modify_target_rbuf_raw(struct char_data *ch, char *rbuf, size_t rbuf_len, in
     buf_mod(rbuf, rbuf_len, "Sunlight", 1);
     WRITEOUT_MSG("Sunlight Allergy", 1);
   }
+  if (IS_GHOUL(ch) && ((time_info.hours > 6) && (time_info.hours < 19)) && OUTSIDE(ch) && weather_info.sky < SKY_RAINING)
+  {
+    base_target += 1;
+    buf_mod(rbuf, rbuf_len, "Sunlight", 1);
+    WRITEOUT_MSG("Sunlight Allergy", 1);
+  }
+  if (GET_RACE(ch) == RACE_DRYAD)
+    if (ROOM_FLAGGED(get_ch_in_room(ch), ROOM_INDOORS && ROOM_ROAD) || SECT(get_ch_in_room(ch)) == SPIRIT_HEARTH && SPIRIT_CITY)
```

Comment:
You'll want a separate ROOM_FLAGGED check for each condition here. Same with SPIRIT_HEARTH vs SPIRIT_CITY.

`if (ROOM_FLAGGED(get_ch_in_room(ch), ROOM_INDOORS) || ROOM_FLAGGED(get_ch_in_room(ch), ROOM_ROAD) || SECT(get_ch_in_room(ch)) == SPIRIT_HEARTH || SECT(get_ch_in_room(ch)) == SPIRIT_CITY)`

You can think of `&&` as meaning "if both the condition on the right and the condition on the left are truthy". When you do `SECT(get_ch_in_room(ch)) == SPIRIT_HEARTH && SPIRIT_CITY`, the condition you're writing is true if "the comparison of sector and spirit hearth is true, AND if spirit_city is truthy". Truthiness in C is defined as not being equal to 0, which is always true since SPIRIT_CITY is really just a human-friendly way of writing `1`.

Reply:
Yeh, my brain struggles with this one for some reason, hopefully this resolves it